File: app/src/main/java/com/hgianastasio/biblioulbrav2/di/modules/ApplicationModule.java
package com.hgianastasio.biblioulbrav2.di.modules;

import android.app.Application;
import android.content.Context;

import java.util.concurrent.Executors;
import java.util.concurrent.ThreadPoolExecutor;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * Created by heitorgianastasio on 4/29/17.
 */
@Module
public class ApplicationModule {
    Application application;

    public ApplicationModule(Application application) {
        this.application = application;
    }

    @Provides @Singleton Context provideContext(){return this.application;}

    @Provides @Singleton
    ThreadPoolExecutor provideExecutor(){return (ThreadPoolExecutor) Executors.newFixedThreadPool(15);}
}

File: app/src/main/java/com/hgianastasio/biblioulbrav2/di/modules/ApplicationModule.kt
package com.hgianastasio.biblioulbrav2.di.modules

import android.app.Application
import android.content.Context
import dagger.Module
import dagger.Provides
import java.util.concurrent.Executors
import java.util.concurrent.ThreadPoolExecutor
import javax.inject.Singleton

/**
 * Created by heitorgianastasio on 4/29/17.
 */
@Module
class ApplicationModule(var application: Application) {
    @Provides
    @Singleton
    fun provideContext(): Context {
        return application
    }

    @Provides
    @Singleton
    fun provideExecutor(): ThreadPoolExecutor {
        return Executors.newFixedThreadPool(15) as ThreadPoolExecutor
    }

}